pkg/objstore/s3/s3.go

```diff
@@ -307,6 +309,53 @@ func (b *Bucket) Name() string {
 	return b.name
 }
 
+// Reads the config file and changes the credentials of the instance if yaml is changed.
+func (b *Bucket) ReloadCredentials() error {
+	if len(b.confFilepath) > 0 {
+		c, err := ioutil.ReadFile(b.confFilepath)
```

Comment:
It might be to costly to do it on every iteration. Is there any asynchronous way to do this?

* We can setup file watch (you can check reloader code)
* Even something simple like checking every 1m.
* Work channels to pass/change the client (????)